aiodocker/multiplexed.py

```diff
@@ -11,30 +11,62 @@ def __init__(self, response):
         self.response = response
 
     async def fetch(self):
-        while True:
-            try:
-                hdrlen = constants.STREAM_HEADER_SIZE_BYTES
-                header = await self.response.content.readexactly(hdrlen)
-                _, length = struct.unpack('>BxxxL', header)
-                if not length:
-                    continue
-                data = await self.response.content.readexactly(length)
-            except (aiohttp.ClientConnectionError,
-                    aiohttp.ServerDisconnectedError):
-                break
-            except asyncio.IncompleteReadError:
-                break
-            yield data.decode('utf8')
+        try:
+            while True:
+                try:
+                    hdrlen = constants.STREAM_HEADER_SIZE_BYTES
+                    header = await self.response.content.readexactly(hdrlen)
+                    _, length = struct.unpack('>BxxxL', header)
+                    if not length:
+                        continue
+                    data = await self.response.content.readexactly(length)
+                except (aiohttp.ClientConnectionError,
+                       aiohttp.ServerDisconnectedError):
+                    break
+                except asyncio.IncompleteReadError:
+                    break
+                yield data.decode('utf8')
+        finally:
+
+            await self.close()
+
+    async def fetch_raw(self):
+        try:
+            async for data in self.response.content.iter_chunked(1024):
+                yield data
+        finally:
+
+            await self.close()
 
     async def close(self):
         await self.response.release()
 
 
-async def multiplexed_result(response, follow=False):
+async def multiplexed_result(response, follow=False, isTty=False):
     log_stream = MultiplexedResult(response)
-    if follow:
-        return log_stream
-    data = []
-    async for record in log_stream.fetch():
-        data.append(record)
-    return data
+
+    if isTty:
+        if follow:
+            return log_stream.fetch_raw()
+        else:
+            d = []
+            decoder = codecs.getincrementaldecoder('utf-8')()
+            async for l in log_stream.fetch_raw():
+                s = decoder.decode(l)
+                d.append(s)
+
+            return ''.join(d)
+    else:
+        if follow:
+            return log_stream.fetch()
+        else:
+            d = []
+            async for l in log_stream.fetch():
+                d.append(l)
+
+            return ''.join(d)
+
+#    data = []
+#    async for record in log_stream.fetch():
+#        data.append(record)
+#    return data
```

Comment:
Please clean up commented codes. Also don't forget rename `isTty` to `is_tty`. :)